docs/blueprints/zombie-segments-cleaner.md

```diff
@@ -0,0 +1,98 @@
+
+Different kind of bad segment is a case where all segments are available on satellite but were deleted from storage nodes.
+
+We should create two cli commands. One for detecting bad segments, second one for deleting segments reported by first comamnd.
+
+The first command for detecting bad segments should iterate over pointerDB and verify each segment. The command should allow specifying flags like DB connection string, and how old segments should be verified. During processing, segment should be assigned to a struct that will collect segments from the same object. All segments from an object where at least one segment is not old enough should be skipped. When the last segment (`l`) will be reached or iteration will be finished each object will be checked if it contains all segments (according to rules from the beginning of this part). If yes, then helper struct will be removed from memory. If not, then all related segments should be printed out. Each entry in such a report should contain segment path, creation/modification date. This command should be **ONLY** executed against pointerDB snapshot to avoid problems with constant changes in the production database.
+
+The second command for deleting bad segments should use the result of first command. The command should allow specifying flags like DB connection string, file with results from the first command, and dry run mode. Each reported segment should be verified if it's not changed since detection report was done. If segment was not changed it should be deleted from pointerDB. Verification and deletion should be done as atomic operation to avoid race conditions. Command should print the report at the end: number of deleted segments, number of segments skipped because of being newer than reported, skipped segments paths. Execution with dry run flag should only print results but without deleting segments from the database. This command should be executed against production database.
+
+## Implementation
+
+Code should be placed in package `cmd/segment-reaper`. The first command for detecting bad segments should be named `detect`. The second command for deleting bad segments should be named `delete`.
+
+Implementation can incorporate `metainfo.PointerDB` and `Iterate` method to go over all segments in DB. 
+
+Proposal for keeping segments structures:
+```
+// key will represent projectID/bucketName/encryptedPath
+map[string]Object
+
+type Object struct {
+    // big.Int represents a bitmask of unlimited size
+    segments big.Int
+    // if skip is true segments from object should be removed from memory when last segment is found 
+    // or iteration is finished,
+    // mark it as true if one of the segments from this object is newer then specified threshold
+    skip     bool 
+}
+```
+
+Output of `segment-reaper detect` command should be CSV with list of segments. Each row will contain segment path and creation/modification date, and serialized pointer for CompareAndSwap function:
```

Comment:
let's output base64, base58 or hex for outputting the path, we can of course, separately add the project id and bucket name without escaping.

The path can contain symbols like "\n\05\08" etc.

Reply:
I also removed note about `;` in csv because with the encoded path we shouldn't have this character